### PR TITLE
Add API for parameter validation

### PR DESCRIPTION
**Pull request by janhohenheim**

### CODE REVIEW

File: Routing/SegmentRegistryFacadeImplementation/RouteRegistrar.cs [hunk 1]
```diff
@@ -17,8 +17,30 @@ public RouteRegistrar(ISegmentParser segmentParser)
 
         public void Register(SegmentNode<TRequest, TResponse> segmentTree, Endpoint endpoint, HandleRequest<TRequest, TResponse> handleRequest)
         {
-            var segments = ParsingUtility.ParseRoute(_segmentParser, endpoint.Route);
-            var targetNode = segments
+            var segments = ParseRoute(endpoint);
+            var targetNode = FindTargetNode(segmentTree, segments);
+            RegisterRequestHandlerOnNode(endpoint, handleRequest, targetNode);
```
Comment (tautropfli) [inline Routing/SegmentRegistryFacadeImplementation/RouteRegistrar.cs]: These two lines are duplicated

File: Routing/SegmentRegistryFacadeImplementation/RouteRegistrar.cs [hunk 2]
```diff
@@ -17,8 +17,30 @@ public RouteRegistrar(ISegmentParser segmentParser)
...
+            Endpoint endpoint,
+            HandleRequest<TRequest, TResponse> handleRequest,
+            ValidateParameters validateParameters)
+        {
+            var segments = ParseRoute(endpoint).ToList();
+
+            RunParameterValidation(validateParameters, segments);
+
+            var targetNode = FindTargetNode(segmentTree, segments);
+            RegisterRequestHandlerOnNode(endpoint, handleRequest, targetNode);
```
Comment (tautropfli) [inline Routing/SegmentRegistryFacadeImplementation/RouteRegistrar.cs:45]: See comment above

**Review by tautropfli — CHANGES_REQUESTED** [review]

File: Routing/SegmentRegistryFacadeImplementation/RouteRegistrar.cs (hunk 2 above)
Comment (janhohenheim) [reply Routing/SegmentRegistryFacadeImplementation/RouteRegistrar.cs:45]: https://github.com/messerli-informatik-ag/routing/pull/12/commits/afdff29a5bddeb40656ccd2b6dad9b8002795329

File: Routing/SegmentRegistryFacadeImplementation/RouteRegistrar.cs (hunk 1 above)
Comment (janhohenheim) [reply Routing/SegmentRegistryFacadeImplementation/RouteRegistrar.cs]: https://github.com/messerli-informatik-ag/routing/pull/12/commits/afdff29a5bddeb40656ccd2b6dad9b8002795329

**Review by tautropfli — APPROVED** [review]

**Pull request merged**